dpd/src/types.rs

```diff
@@ -85,6 +85,10 @@ pub enum DpdError {
     Oximeter(String),
     #[error("No switch identifiers available")]
     NoSwitchIdentifiers,
+    #[error("Multicast group error: {}", .0)]
+    McastGroup(String),
+    #[error("Resource exhausted: {}", .0)]
```

Comment:
The only place this is used is when we're out of group IDs.  You could just make that a specific error, rather than the vague "resource exhausted".

Reply:
ResourceExhaused is for the next line, but I'll rename this to something more apt.